### PR TITLE
Update CI tests

### LINKED ISSUE #119: Githhub Actions version
Hi 👋 I had a few questions regarding the CI tests:

Is there a reason not to test python 3.9 in CI tests? https://github.com/drvinceknight/Nashpy/blob/main/.github/workflows/config.yml

Also, instead of calling tox like this:
`python -m tox -e py38`
We can set up tox-gh-actions https://github.com/ymyzk/tox-gh-actions, to automatically assign tox environments according to the python version and simply call:
`python -m tox`

And lastly, is there any particular reason to use actions/checkout@v1? v2 is now stable and can be used safely. (Same for setup-python)

### PR DESCRIPTION
**Pull request by asinghgaba**
Fixes #119 

Now added support for Python 3.9 and uses tox-gh-actions.
Also changed the version for checkout and setup-python.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -2,6 +2,11 @@
 isolated_build = True
 envlist = py38, py39
 
+[gh-actions]
+python =
```
Comment (drvinceknight) [inline tox.ini:6]: This all looks fantastic, thank you. What is this particular thing for @asinghgaba ?

**Comment by drvinceknight** [comment]
Thanks for this! :+1:

File: tox.ini (hunk 1 above)
Comment (asinghgaba) [reply tox.ini:6]: This tells which tox environment to use for the particular python job when running the workflow on Github actions.

Comment (drvinceknight) [reply tox.ini:6]: Is it specifically used by `tox-gh-actions`?

Comment (asinghgaba) [reply tox.ini:6]: yes exactly!

Comment (drvinceknight) [reply tox.ini:6]: Awesome, thanks for explaining that.

**Pull request merged**